src/Utils/BlockTemplateUtils.php

```diff
@@ -74,6 +74,143 @@ public static function gutenberg_inject_theme_attribute_in_content( $template_co
+		if ( ! \WP_Theme_JSON_Resolver_Gutenberg::theme_has_support() ) {
+			return $template_item;
+		}
+
+		$theme_data = \WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_custom_templates();
+		if ( isset( $theme_data[ $template_item['slug'] ] ) ) {
+			$template_item['title']     = $theme_data[ $template_item['slug'] ]['title'];
+			$template_item['postTypes'] = $theme_data[ $template_item['slug'] ]['postTypes'];
+		}
+
+		return $template_item;
+	}
+
+	/**
+	 * Retrieves the template file from the theme for a given slug.
+	 *
+	 * @internal
+	 *
+	 * @param string $template_type wp_template or wp_template_part.
+	 * @param string $slug template slug.
+	 *
+	 * @return array|null Template.
+	 */
+	public static function gutenberg_get_template_file( $template_type, $slug ) {
+		$template_base_paths = array(
+			'wp_template'      => 'block-templates',
+			'wp_template_part' => 'block-template-parts',
+		);
+		$themes              = array(
+			get_stylesheet() => get_stylesheet_directory(),
+			get_template()   => get_template_directory(),
+		);
+		foreach ( $themes as $theme_slug => $theme_dir ) {
+			$file_path = $theme_dir . '/' . $template_base_paths[ $template_type ] . '/' . $slug . '.html';
+			if ( file_exists( $file_path ) ) {
+				$new_template_item = array(
+					'slug'  => $slug,
+					'path'  => $file_path,
+					'theme' => $theme_slug,
+					'type'  => $template_type,
+				);
+
+				if ( 'wp_template_part' === $template_type ) {
+					return self::gutenberg_add_template_part_area_info( $new_template_item );
+				}
+
+				if ( 'wp_template' === $template_type ) {
+					return self::gutenberg_add_template_info( $new_template_item );
+				}
+
+				return $new_template_item;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Attempts to add the template part's area information to the input template.
+	 *
+	 * @param array $template_info Template to add information to (requires 'type' and 'slug' fields).
+	 *
+	 * @return array Template.
+	 */
+	public static function gutenberg_add_template_part_area_info( $template_info ) {
+		if ( \WP_Theme_JSON_Resolver_Gutenberg::theme_has_support() ) {
+			$theme_data = \WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_template_parts();
+		}
+
+		if ( isset( $theme_data[ $template_info['slug'] ]['area'] ) ) {
+			$template_info['title'] = $theme_data[ $template_info['slug'] ]['title'];
+			$template_info['area']  = gutenberg_filter_template_part_area( $theme_data[ $template_info['slug'] ]['area'] );
+		} else {
+			$template_info['area'] = WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
+		}
+
+		return $template_info;
+	}
+
+	/**
+	 * Build a unified template object based a post Object.
+	 *
+	 * @param \WP_Post $post Template post.
+	 *
+	 * @return \WP_Block_Template|\WP_Error Template.
+	 */
+	public static function gutenberg_build_template_result_from_post( $post ) {
+		$default_template_types = gutenberg_get_default_template_types();
```

Comment:
No need for this once its usage below has been removed so we can remove this also